### PR TITLE
Feat/history single video

### PR DESCRIPTION
**Pull request by srishti-maurya**
## Added
 - single video page
 - like video
 - history
 - category filter

### CODE REVIEW

**Review by preethasri — APPROVED** [review]
good to go

File: src/frontend/Components/VideoCard.jsx [hunk 1]
```diff
@@ -2,24 +2,30 @@ import { useState } from "react";
...
 import { useData } from "../Contexts/data-context";
+import { useAuth } from "../Contexts/auth-context";
 
 export function getThumbnail(id) {
   return `https://i.ytimg.com/vi/${id}/hq720.jpg`;
 }
+
 export function getCreatorImg(id) {
   return `https://yt3.ggpht.com/ytc/${id}`;
 }
```
Comment (gopalbbharadva) [inline src/frontend/Components/VideoCard.jsx:19]: Suggestion: you can put these functions in utils file and import it here.

File: src/frontend/Contexts/data-context.js [hunk 2]
```diff
@@ -13,6 +7,56 @@ const DataContext = createContext();
...
+  ) {
+    // console.log(data);
+    return data;
+  }
+
+  if (AUDIOBOOK) {
+    return data.filter((item) => item.category === "audiobook");
+    // let newList = data.filter((item) => item.category === "audiobook");
+    // filteredlist.push(...newList);
+  }
```
Comment (gopalbbharadva) [inline src/frontend/Contexts/data-context.js:31]: u can remove these comments if not needed

File: src/frontend/Contexts/data-context.js [hunk 3]
```diff
@@ -13,6 +7,56 @@ const DataContext = createContext();
...
+
+  if (BENEFITS) {
+    return data.filter((item) => item.category === "benefits");
+    // let newList = data.filter((item) => item.category === "benefits");
+    // filteredlist.push(...newList);
+  }
+  if (ALL_CATEGORY) {
+    return data;
+    // filteredlist = data;
+  }
```
Comment (gopalbbharadva) [inline src/frontend/Contexts/data-context.js:46]: u can remove these comments if not needed

**Review by gopalbbharadva — APPROVED** [review]
Functionality working fine and code looks very good. Good to go.

**Pull request merged**